oci/client.go

```diff
@@ -36,3 +36,8 @@ type Image interface {
 	// ContentStore provides a content store which contains image blob data
 	ContentStore() content.Store
 }
+
+// NameSpace interface user by some SpecOpt to get client default namespace
+type NameSpace interface {
```

Comment:
What is this interface for?

Reply:
If the `WithNamespace` is moved into the root package, there is no need for this interface or the `Namespace()` function. This interface would only be necessary if it was being used in the `oci` package.

Reply:
It seems that moving `WithNamespace` to the root package can also solve this problem, currently I use `WithNewSpec` to inject client default namespace.